src/v/kafka/server/handlers/metadata.cc

```diff
@@ -158,15 +158,17 @@ create_topic(request_context& ctx, model::topic&& topic) {
               metadata_response::topic t;
               t.name = std::move(res[0].tp_ns.tp);
               t.error_code = map_topic_error_code(res[0].ec);
-              return ss::make_ready_future<metadata_response::topic>(t);
+              return ss::make_ready_future<metadata_response::topic>(
+                std::move(t));
```

Comment:
oh good catch! sorta surprised we couldn't do `return t` and get NRVO here.

Reply:
Well we need to wrap it in the ready future and that will suppress NRVO, right?

Reply:
i had some thought about how seastar had managed to optimize this case, but now i cannot find this reference. i agree it would be surprising.

Reply:
@dotnwat - I think when you return a bare value `T` inside a continuation, it gets converted by seastar into a `future<T>` but can still use NVRO, becuase seastar jumps thru some hoops to ensure that works: maybe that's what you were thinking of?